best-practice-certificate-provisioning.md

```diff
@@ -0,0 +1,285 @@
+
+## EST Server Behaviour
+
+The EST server must be implemented in accordance with [RFC 7030][RFC-7030].
+
+### EST Server API
+
+The EST Server SHALL present an instance of the EST API as defined in [RFC 7030][RFC-7030].
+
+This MUST include the following API endpoints:
+
+| Operation                       | Operation path  |
+| ------------------------------- | --------------- |
+| Distribution of CA Certificates | /cacerts        |
+| Enrollment of Clients           | /simpleenroll   |
+| Re-enrollment of Clients        | /simplereenroll |
+
+
+### EST Server Authentication
+
+The EST Server MUST present a valid TLS Server Certificate signed by the CA for realm from which it is issuing certificates.
+
+### EST Server - Client Authentication
+
+The EST Server MUST authenticate the EST Client that is requesting a TLS Certificate manually or automatically using a TLS Client Certificate.
+
+The EST Server MUST support using a TLS Client Certificate, presented during the TLS handshake by the EST Client to authenticate if the Client is trusted. The TLS Client Certificate can either be signed by the current CA or an externally trusted Root CA. The EST Server MUST check the validity of the EST CLients TLS Certificate before responding to its request. The EST Server MUST provide a method to load multiple trusted Root CA's, that are used to verify TLS Client Certificate.
+
+The EST Server MAY also support manually authentication of the EST Client if; no TLS Client Certificate is presented during the TLS handshake, the TLS Client Certificate is not trusted or as an extra authentication step. The exact process for manual authentication will be implementation specific, but the EST Server MUST provide enough information to the operator so they can authenticate the EST Client. During the manual authentication the EST Server MUST respond with either HTTP 202 or HTTP 503 and the response must include a `Retry-After` header.
+
+The EST Server MUST be capable of issuing TLS Certificates signed with both RSA and ECDSA keys.
+
+The EST Server MUST return a TLS Certificate with the Extended Key Usage set for both TLS Server Authentication and TLS Client AUthentication as per [RFC 5280][RFC-5280]. This is to allow the TLS Certificate to be used to authenticated the NMOS APIs to NMOS Clients and for authentication of the EST Client during certificate renewal.
+
+## EST Client
+
+An EST Client SHOULD allow EST to be disabled, preventing the EST Client from being automatically provisioned with a TLS Certificate if required by the networks security policy.
+
+An EST Client SHOULD allow manual configuration of the EST Servers Hostname and Port, to prevent the EST Client from requesting a TLS Certificate from a rogue server.
+
+An EST Client MUST provide a method to manually install both the Root CA and TLS Server certificate for the target environment, for the case when an EST Server is not present or the TLS Client Certificate is no longer valid.
```

Comment:
Presumably there must also be a way to replace the installed TLS client cert in case it is revoked or expires?

Reply:
We should also provide guidance here on the expected lifetime of the manufacturer installed cert and its associated CA.